service/src/main/java/uk/gov/hmcts/reform/fpl/handlers/FurtherEvidenceUploadedEventHandler.java

```diff
@@ -0,0 +1,73 @@
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class FurtherEvidenceUploadedEventHandler {
+    private final FurtherEvidenceNotificationService furtherEvidenceNotificationService;
+
+    @EventListener
+    public void handleDocumentUploadedEvent(final FurtherEvidenceUploadedEvent event) {
+        final CaseData caseData = event.getCaseData();
+        final CaseData caseDataBefore = event.getCaseDataBefore();
+        final String excludedEmail = event.getInitiatedBy().getEmail();
+        final String sender = event.getInitiatedBy().getFullName();
+
+        if (event.isUploadedByLA()) {
+            if (hasNewNonConfidentialDocuments(caseData.getFurtherEvidenceDocumentsLA(),
+                caseDataBefore.getFurtherEvidenceDocumentsLA())) {
+                notifyRespondents(caseData, excludedEmail, sender);
+            }
+        } else {
+            if (hasNewNonConfidentialDocuments(caseData.getFurtherEvidenceDocuments(),
+                caseDataBefore.getFurtherEvidenceDocuments())) {
+                notifyLASolicitors(caseData, excludedEmail, sender);
+                notifyRespondents(caseData, excludedEmail, sender);
+            }
+        }
+    }
+
+    private void notifyLASolicitors(final CaseData caseData, final String excludeEmail, final String sender) {
+        Set<String> recipients = furtherEvidenceNotificationService.getLocalAuthoritySolicitorEmails(caseData);
+        furtherEvidenceNotificationService.sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData, filterRecipients(recipients, excludeEmail), sender);
+    }
+
+    private void notifyRespondents(final CaseData caseData, final String excludeEmail, final String sender) {
+        Set<String> recipients = furtherEvidenceNotificationService.getRespondentRepresentativeEmails(caseData);
+        furtherEvidenceNotificationService.sendFurtherEvidenceDocumentsUploadedNotification(
+            caseData, filterRecipients(recipients, excludeEmail), sender);
+    }
+
+    private static boolean hasNewNonConfidentialDocuments(List<Element<SupportingEvidenceBundle>> newEvidenceBundle,
+                                                          List<Element<SupportingEvidenceBundle>> oldEvidenceBundle) {
+        List<SupportingEvidenceBundle> oldEvidenceBundleUnwrapped = unwrapElements(oldEvidenceBundle);
+        return unwrapElements(newEvidenceBundle).stream()
+            .anyMatch(d -> oldEvidenceBundleUnwrapped.stream()
+                .noneMatch(old -> old.getDocument().equals(d.getDocument()))
+                && !d.isConfidentialDocument());
+    }
+
+    private static Set<String> filterRecipients(final Set<String> recipients, final String excludeEmail) {
+        return StringUtils.isEmpty(excludeEmail) ? recipients :
```

Comment:
can be just 
```
       return recipients.stream()
            .filter(r -> !r.equals(excludeEmail))
            .collect(Collectors.toSet());
```
as it will work with null excluded as well. In theory excludeEmail should never be null as this is current user email and cannot be null. 
Other option is just recipients.remove(excludeEmail) but it work only with mutable collections